### PR TITLE
Add condition to check last rejected diff

### LINKED ISSUE #70: Check the old rejected diff
Check the old rejected Diff in the profile function if that is similar to the new Diff doesn't store the profileDiff.

Diffs are checked here: https://github.com/Real-Dev-Squad/identity-service/blob/1084d95a1aa3ed44536bd5fd1acfed0c86ebd882/profile/main.go#L287

### PR DESCRIPTION
**Pull request by shubham-y**
Closes #70

### CODE REVIEW

File: profile/main.go [hunk 1]
```diff
@@ -282,16 +282,19 @@ func getdata(client *firestore.Client, ctx context.Context, userId string, userU
...
-		if lastdiffId != "" {
-			setNotApproved(client, ctx, lastdiffId)
+	if lastPendingDiff != res && userData != res {
+		if lastPendingDiffId != "" {
+			setNotApproved(client, ctx, lastPendingDiffId)
+		}
+		lastRejectedDiff, _ := getLastDiff(client, ctx, userId, "NOT APPROVED")
+		if lastRejectedDiff != res {
+			generateAndStoreDiff(client, ctx, res, userId)
 		}
```
Comment (lakshayman) [inline profile/main.go:294]: let's log it in the else part with
type: "SAME_PROFILE_DIFF"
timestamp: <timestamp>
meta: { profileDiffId: <lastDiffRejected> }
body: { profileDiffId: <lastDiffRejected> }

Comment (shubham-y) [reply profile/main.go:315]: Changes are done

File: profile/main.go [hunk 2]
```diff
@@ -181,6 +181,25 @@ func logHealth(client *firestore.Client, ctx context.Context, userId string, isS
 	client.Collection("logs").Add(ctx, newLog)
 }
 
+/*
+ Logs the status of the user's profileDiff
+*/
+func logProfileDiffStatus(client *firestore.Client, ctx context.Context, userId string, profileDiffId string) {
```
Comment (lakshayman) [inline profile/main.go:187]: Can we please give it a bit relevant name like logSameProfileDiffGenerated

Comment (shubham-y) [reply profile/main.go]: Good point. I have made the change

**Review by lakshayman — APPROVED** [review]
LGTM

**Pull request merged**